src/settings/settings.go

```diff
@@ -106,6 +106,7 @@ type Settings struct {
 	CacheKeyPrefix                     string  `envconfig:"CACHE_KEY_PREFIX" default:""`
 	BackendType                        string  `envconfig:"BACKEND_TYPE" default:"redis"`
 	StopCacheKeyIncrementWhenOverlimit bool    `envconfig:"STOP_CACHE_KEY_INCREMENT_WHEN_OVERLIMIT" default:"false"`
+	HitsAddendMinValue                 uint32  `envconfig:"DEFAULT_HITS_ADDEND_MIN_VALUE" default:"1"`
```

Comment:
can we do w/o this extra field ?
we'll need to make HitsAddend optional in the API https://github.com/envoyproxy/go-control-plane/blob/e0e383246f4cd59a40967987057c76f03526cffb/envoy/service/ratelimit/v3/rls.pb.go#L179
which may be a breaking change for clients
cc @zirain @mathetake @wbpcode

Reply:
if this still defaults to 1 then i don't think this would be a breaking change in practice i guess? but ack we should need to allow zero in the API regardless

Reply:
@arkodg I confirm that this is not a breaking change and that the `HitsAddendMinValue` field introduced here is not needed in the `RatelimitRequest` request send to the ratelimit service.

This is only a config variable of the reatelimit service, with default value of 1 that matches the current **hardcoded** value of 1.

@mathetake there is no change to be done to the API, as zero is already a valid value for the `HitsAddend` value. In fact it is the **default value** when no custom `HitsAddend` value is set via the [Set-Filter-State](https://www.envoyproxy.io/docs/envoy/latest/configuration/http/http_filters/set_filter_state) HTTP filter or directly in the `RatelimitRequest` and the actual `+1` increment is enforced at the ratelimit service level (if the hitAddend value is set to zero in the request -- either voluntarily or because it is absent in the request-- the value is [override to `1`](https://github.com/envoyproxy/ratelimit/blob/28b1629a21e885bdd2b527d6a1c1de8483dc47d4/src/redis/fixed_cache_impl.go#L52). 

Currently, if someone uses the  `FilterState` to enforce the hitsAddend value **voluntarily** to zero (we have a use case for that) or set the `HitsAddend` value to zero in the `RatelimitRequest`, then in practice the limit counter will still be incremented because of the `max(request.HitsAddend, 1)` check that is done, which is not expected. 

What this PR does basically is just to allow to change this hardcoded `1` to any value `>=0` ([same constraints than the api](https://github.com/EItanya/envoy/blob/b1a2af6a29b6a6d95449d407c9e3e5ae64c74788/api/envoy/service/ratelimit/v3/rls.proto#L50), but still keeping 1 as default value to not change the actual behavior) to allow `RatelimitRequests` with zero increment.

Reply:
hey @gcalmettes, we understand the issue and our use case is the same as yours :)
I was questioning the current RLS API and the semantics defined in [RateLimitRequest](https://github.com/envoyproxy/go-control-plane/blob/e0e383246f4cd59a40967987057c76f03526cffb/envoy/service/ratelimit/v3/rls.pb.go#L163C6-L163C22) , since its not a pointer, there's no way to figure out unset (default:1 ) vs set to 0.
It maybe too late to edit that API, the easiest thing to do maybe is 
* set the default value to `1` in envoy if `HitsAddend` is not set
* update the code in this project to `max(request.HitsAddend, 0)` simplifying the logic and maintaining backwards compatibility
wdyt @mathetake @wbpcode

Reply:
sounds like you had the same idea @mathetake https://github.com/envoyproxy/ratelimit/pull/729#issuecomment-2539365446